src/pages/signin/ValidateCodeForm/BaseValidateCodeForm.js

```diff
@@ -63,224 +64,217 @@ const defaultProps = {
-        this.setState({linkSent: true});
-    }
+        setLinkSent(true);
+    };
 
     /**
      * Clears local and Onyx sign in states
      */
-    clearSignInData() {
-        this.setState({twoFactorAuthCode: '', formError: {}, validateCode: ''});
+    const clearSignInData = () => {
+        setTwoFactorAuthCode('');
+        setFormError({});
+        setValidateCode('');
         Session.clearSignInData();
-    }
+    };
 
     /**
      * Check that all the form fields are valid, then trigger the submit callback
      */
-    validateAndSubmitForm() {
-        const requiresTwoFactorAuth = this.props.account.requiresTwoFactorAuth;
-
+    const validateAndSubmitForm = useCallback(() => {
+        const requiresTwoFactorAuth = props.account.requiresTwoFactorAuth;
         if (requiresTwoFactorAuth) {
-            if (this.input2FA) {
-                this.input2FA.blur();
+            if (input2FARef.current) {
+                input2FARef.current.blur();
             }
-
-            if (!this.state.twoFactorAuthCode.trim()) {
-                this.setState({formError: {twoFactorAuthCode: 'validateCodeForm.error.pleaseFillTwoFactorAuth'}});
+            if (!twoFactorAuthCode.trim()) {
+                setFormError({twoFactorAuthCode: 'validateCodeForm.error.pleaseFillTwoFactorAuth'});
                 return;
             }
-
-            if (!ValidationUtils.isValidTwoFactorCode(this.state.twoFactorAuthCode)) {
-                this.setState({formError: {twoFactorAuthCode: 'passwordForm.error.incorrect2fa'}});
+            if (!ValidationUtils.isValidTwoFactorCode(twoFactorAuthCode)) {
+                setFormError({twoFactorAuthCode: 'passwordForm.error.incorrect2fa'});
                 return;
             }
         } else {
-            if (this.inputValidateCode) {
-                this.inputValidateCode.blur();
+            if (inputValidateCodeRef.current) {
+                inputValidateCodeRef.current.blur();
             }
-
-            if (!this.state.validateCode.trim()) {
-                this.setState({formError: {validateCode: 'validateCodeForm.error.pleaseFillMagicCode'}});
+            if (!validateCode.trim()) {
+                setFormError({validateCode: 'validateCodeForm.error.pleaseFillMagicCode'});
                 return;
             }
-            if (!ValidationUtils.isValidValidateCode(this.state.validateCode)) {
-                this.setState({formError: {validateCode: 'validateCodeForm.error.incorrectMagicCode'}});
+            if (!ValidationUtils.isValidValidateCode(validateCode)) {
+                setFormError({validateCode: 'validateCodeForm.error.incorrectMagicCode'});
                 return;
             }
         }
+        setFormError({});
 
-        this.setState({
-            formError: {},
-        });
-
-        const accountID = lodashGet(this.props, 'credentials.accountID');
+        const accountID = lodashGet(props, 'credentials.accountID');
```

Comment:
```suggestion
        const accountID = lodashGet(props.credentials, 'accountID');
```

To avoid adding `props` as a dependency we can take a shortcut to `props.credentials`. `props.credentials` is always set (or have a default value) so it's safe to access.